### PR TITLE
bench: add custom workload

### PR DESCRIPTION
**Pull request by gabriele-0201**
from the cli, it is now possible to execute workloads like:
set_balace (heavy_write), heavy_read, heavy_update, or heavy_delete

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/122?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#122** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/122?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#114** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/114?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#109** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/109?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#116** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/116?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#108** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/108?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#107** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/107?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#86** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/86?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gabriele-0201 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: xtask/src/workload.rs [hunk 1]
```diff
@@ -51,38 +29,38 @@ pub fn parse(
             percentage_cold_transfer: percentage_cold_transfer.unwrap_or(0),
             additional_initial_capacity,
         }),
-        "set_balance" => Box::new(CustomWorkload {
-            reads: 0,
-            writes: 100,
-            delete: 0,
-            update: 0,
+        "set_balance" | "heavy_write" => Box::new(CustomWorkload::new(
```
Comment (rphmeier) [inline xtask/src/workload.rs:32]: why are these the same?

File: xtask/src/workload.rs [hunk 2]
```diff
@@ -51,38 +29,38 @@ pub fn parse(
...
+            0,   // reads
+            100, // writes
```
Comment (rphmeier) [inline xtask/src/workload.rs:34]: The old constructor looked like it was avoiding these comments - this seems like a step back

File: xtask/src/workload.rs (hunk 1 above)
Comment (gabriele-0201) [reply xtask/src/workload.rs:32]: A week ago, during a discussion, it emerged that the two main workloads could have been transfer and set_balance, but set_balance is just a bunch of writes into the storage (if I'm not mistaken). Thus, having two separate things, set_balance and the custom workload, seems useless, so I just named set_balance as the default workload.

Anyway, the default workloads will likely become easier, such as `randr`, `randw`, etc., and so those names will disappear

File: xtask/src/workload.rs (hunk 2 above)
Comment (gabriele-0201) [reply xtask/src/workload.rs:34]: Agree, but having a constructor that checks on percentage is a step forward for the feature of creating completely custom workloads

**Pull request merged**